src/components/ModelSelector.jsx

```diff
@@ -7,31 +7,37 @@ import {
   ToggleButton,
 } from "react-bootstrap";
 
-import { useSelectorContext } from "../context/selector-context";
-import { colorMaps, measurement, season, tide } from "../assets/config.json";
-import { useControlsContext } from "../context/controls-context";
-
-export default function ModelSelector({ toggleControls }) {
+export default function ModelSelector(props) {
   const {
-    state: { selection },
-    dispatch: selectorDispatch,
-  } = useSelectorContext();
+    BUTTONS,
+    ALL_COLOR_MAPS,
+    selection,
+    setSelection,
+    setColorMap,
+    toggleControls,
+  } = props;
 
-  const { dispatch: controlsDispatch } = useControlsContext();
+  function handleMeasurementChange(val) {
+    setSelection({
+      ...selection,
+      measurement: val,
+    });
+    setColorMap(
+      val.value === "salt" ? ALL_COLOR_MAPS.Haline : ALL_COLOR_MAPS.Thermal
+    );
+  }
 
-  function handleChange(val) {
-    // Change model
-    selectorDispatch({
-      type: "TOGGLE_MEASUREMENT",
-      payload: val,
+  function handleSeasonChange(val) {
+    setSelection({
+      ...selection,
+      season: val,
     });
-    // Change color map
-    controlsDispatch({
-      type: "CHANGE_COLOR_MAP",
-      payload:
-        val.value === "salt"
-          ? colorMaps.find((m) => m.name === "Haline")
-          : colorMaps.find((m) => m.name === "Thermal"),
+  }
+
+  function handleTideChange(val) {
+    setSelection({
+      ...selection,
+      tide: val,
```

Comment:
maybe make all of these an object with the tide/season/measurement as keys and the values as these functions, then you can make the variables a map in the jsx as well instead of having repetitive code there

Reply:
I did the map based off `BUTTONS`